gammapy/maps/hpxsparse.py

```diff
@@ -62,15 +102,46 @@ def fill_by_idx(self, idx, weights=None):
         if weights is None:
             weights = np.ones(idx[0].shape)
         idx = self.hpx.global_to_local(idx)
-        idx = ravel_hpx_index(idx, self.hpx.npix)
-        self.data[0, idx] += weights
+        idx_flat = np.ravel_multi_index(idx, self.data.shape[::-1])
+        idx_flat, idx_inv = np.unique(idx_flat, return_inverse=True)
+        idx = np.unravel_index(idx_flat, self.data.shape[::-1])
+        weights = np.bincount(idx_inv, weights=weights)
+        self.data.set(idx[::-1], weights, fill=True)
 
     def set_by_idx(self, idx, vals):
 
         idx = pix_tuple_to_idx(idx)
         idx = self.hpx.global_to_local(idx)
-        idx = ravel_hpx_index(idx, self.hpx.npix)
-        self.data[0, idx] = vals
+        self.data[idx[::-1]] = vals
+
+    def make_cols(self, header, conv):
+
+        from astropy.io import fits
```

Comment:
Same here: please move the import to the top, and add a docstring to the `make_cols` function. At least a one-line docstring stating the purpose of the method.